src/cryptoadvance/specter/util/tor.py

```diff
@@ -59,3 +62,9 @@ def stop_hidden_services(app):
             app.tor_service_id = None
     except Exception:
         pass  # we tried...
+
+def renew_tor_ip():
+    with Controller.from_port(port=9051) as controller:
+        controller.authenticate(password='')
+        controller.signal(Signal.NEWNYM)
+        time.sleep(secrets.SystemRandom().randint(15, 30)) # Ensure new IP used
```

Comment:
So this means there will be a 15-30 second delay for each utxo?

Reply:
Yes, too much I guess? What do you think will be best, maybe none?

Reply:
It's ok as soon as we show that something is still happening.
So I assume we need a new progress status that utxo rescan is done and we are fetching data over tor now.
At the moment the user only sees that rescan is done but nothing appeared in the wallet.

Reply:
Just checked, for me it either changes IP address without any delay or doesn't change it at all.